File: app/todo/ui/TrackActivity.java
//
// Translated by CS2J (http://www.cs2j.com): 2014-07-29 22:02:12
//

package dk.aschoen.beatplanner.ui;






public class TrackActivity  extends Activity 
{
    private Metronome __Metronome;
    public Metronome getMetronome() {
        return __Metronome;
    }

    public void setMetronome(Metronome value) {
        __Metronome = value;
    }

    private List<SequenceView> seqViews = new List<SequenceView>();
    protected void onCreate(Bundle bundle) throws Exception {
        super.OnCreate(bundle);
        setMetronome(MetroApp.DefaultMetronome(this));
        SetContentView(Resource.Layout.Track);
        /* [UNSUPPORTED] 'var' as type is unsupported "var" */ cv = FindViewById<LinearLayout>(Resource.Id.trackRoot);
        for (int i = 0;i < cv.ChildCount;i++)
        {
            View v = cv.GetChildAt(i);
            if (v instanceof SequenceView)
                addSequenceView((SequenceView)v);
             
        }
    }

    //      var root = FindViewById<LinearLayout>(Resource.Id.trackRoot);
    //      view_sequence sv = new view_sequence(this, new Sequence(new beat(Meter.Common, 100), 4));
    //      root.AddView(sv);
    private void addSequenceView(SequenceView v) throws Exception {
        v.getPlayBtn().Click += /* [UNSUPPORTED] to translate lambda expressions we need an explicit delegate type, try adding a cast "(Object sender, EventArgs e) => {
            Sequence s = v.Sequence;
            /* [UNSUPPORTED] 'var' as type is unsupported "var" */ btn = (Button)sender;
            getMetronome().setBeat(s.Beat);
            getMetronome().toggle();
            btn.Text = (StringSupport.equals(btn.Text, "Play")) ? "Stop" : "Play";
        }" */;
        seqViews.Add(v);
    }

}

File: BeatPlannerAndroid/src/beatplanner/ui/activities/TrackActivity.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System.Security.Cryptography;
using BeatPlanner;

namespace BeatPlannerAndroid
{
  [Activity(Label = "TrackActivity", MainLauncher = true, Icon = "@drawable/icon")]			
  public class TrackActivity : Activity
  {

    public Metronome Metronome { get; private set; }

    private List<SequenceView> seqViews = new List<SequenceView>();

    protected override void OnCreate(Bundle bundle)
    {
      base.OnCreate(bundle);

      Metronome = MetroApp.DefaultMetronome(this);

      SetContentView(Resource.Layout.Track);


      var cv = FindViewById<LinearLayout>(Resource.Id.trackRoot);

      for (int i = 0; i < cv.ChildCount; i++)
      {
        View v = cv.GetChildAt(i);
        if (v is SequenceView)
          AddSequenceView((SequenceView)v);
      }

//      var root = FindViewById<LinearLayout>(Resource.Id.trackRoot);
//      SequenceView sv = new SequenceView(this, new Sequence(new Beat(Meter.Common, 100), 4));
//      root.AddView(sv);
    }

    private void AddSequenceView(SequenceView v)
    {
      v.PlayBtn.Click += (object sender, EventArgs e) =>
      {
        var s = v.Sequence;
        var btn = (Button)sender;
        Metronome.Beat = s.Beat;
        Metronome.Toggle();
        btn.Text = (btn.Text == "Play") ? "Stop" : "Play";

      };

      seqViews.Add(v);
    }
  }
}

